pkg/genall/output.go

```diff
@@ -155,6 +155,12 @@ func (o OutputArtifacts) Open(pkg *loader.Package, itemPath string) (io.WriteClo
 		return nil, fmt.Errorf("cannot output to a package with no path on disk")
 	}
 	outDir := filepath.Dir(pkg.CompiledGoFiles[0])
+	if _, err := os.Stat(outDir); os.IsNotExist(err) {
```

Comment:
Nit: Why was this added?

Reply:
The apply configurations need to be generated in a new package from the provided types. Right now the plan is to use "{old-pkg-path}/ac" to represent the new package path. This will require creating a new directory. I'm currently doing this by modifying the pkg.CompiledGoFiles[0] so that the output lives in the right directory (https://github.com/kubernetes-sigs/controller-tools/blob/fde36c5017ec5217b8e0b22e8e9c941cc2678a72/pkg/applyconfigurations/gen.go#L116). Not exactly sure if this is the best way to do it, let me know if you have suggestions.